libs/text-splitters/langchain_text_splitters/base.py

```diff
@@ -151,20 +151,23 @@ def from_huggingface_tokenizer(cls, tokenizer: Any, **kwargs: Any) -> TextSplitt
         """Text splitter that uses HuggingFace tokenizer to count length."""
         try:
             from transformers import PreTrainedTokenizerBase
-
-            if not isinstance(tokenizer, PreTrainedTokenizerBase):
-                raise ValueError(
-                    "Tokenizer received was not an instance of PreTrainedTokenizerBase"
-                )
-
-            def _huggingface_tokenizer_length(text: str) -> int:
-                return len(tokenizer.encode(text))
-
         except ImportError:
             raise ValueError(
                 "Could not import transformers python package. "
                 "Please install it with `pip install transformers`."
             )
+
+        if not isinstance(tokenizer, PreTrainedTokenizerBase):
+            raise ValueError(
+                "Tokenizer received was not an instance of PreTrainedTokenizerBase"
+            )
+
+        def _huggingface_tokenizer_length(text: str) -> int:
+            return len(tokenizer.encode(text))
+
+        if issubclass(cls, TokenTextSplitter):
```

Comment:
This isn't the correct place for the code. Code inside of `TextSplitter` should not be aware of the existence of `TokenTextSplitter`. The way to do this would be to override the classmethod if TokenTextSplitter needs some special behavior.

Reply:
Yeah that confused me as well. I feel like from_huggingface_tokenizer and from_tiktoken_encoder should be class methods for TokenTextSplitter